src/pe/header.rs

```diff
@@ -396,73 +407,300 @@ impl Default for DosStub {
+    /// * [`COFF_MACHINE_M32R`],
+    /// * [`COFF_MACHINE_MIPS16`],
+    /// * [`COFF_MACHINE_MIPSFPU`],
+    /// * [`COFF_MACHINE_MIPSFPU16`],
+    /// * [`COFF_MACHINE_POWERPC`],
+    /// * [`COFF_MACHINE_POWERPCFP`],
+    /// * [`COFF_MACHINE_R4000`],
+    /// * [`COFF_MACHINE_RISCV32`],
+    /// * [`COFF_MACHINE_RISCV64`],
+    /// * [`COFF_MACHINE_RISCV128`],
+    /// * [`COFF_MACHINE_SH3`],
+    /// * [`COFF_MACHINE_SH3DSP`],
+    /// * [`COFF_MACHINE_SH4`],
+    /// * [`COFF_MACHINE_SH5`],
+    /// * [`COFF_MACHINE_THUMB`],
+    /// * [`COFF_MACHINE_WCEMIPSV2`],
+    ///
+    /// or any other value that is not listed here.
+    ///
+    /// The constants above are sourced from <https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#machine-types>.
+    /// If there's a missing constant, please open an issue or a pull request.
+    // TODO: insert the values names with a macro
+    #[doc(alias("Machine"))]
     pub machine: u16,
+    /// The number of sections. This indicates the size of the section table, which immediately follows the headers.
+    /// Note that the Windows loader limits the number of sections to 96.
+    /// [Source](https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-image_file_header).
+    #[doc(alias("NumberOfSections"))]
     pub number_of_sections: u16,
+    /// The low 32 bits of the time stamp of the image. This represents the date and time the image was created by the linker.
+    /// The value is represented in the number of seconds elapsed since midnight (00:00:00), January 1, 1970, Universal
+    /// Coordinated Time, according to the system clock.
+    #[doc(alias("TimeDateStamp"))]
     pub time_date_stamp: u32,
+    /// The offset of the symbol table, in bytes, or zero if no COFF symbol table exists.
+    ///
+    /// Typically, this field is set to 0 because COFF debugging information is deprecated.
+    /// [Source](https://0xrick.github.io/win-internals/pe4/#file-header-image_file_header).
+    // TODO: further explain the COFF symbol table. This seems to be a nuanced topic.
+    #[doc(alias("PointerToSymbolTable"))]
     pub pointer_to_symbol_table: u32,
+    /// The number of symbols in the symbol table.
+    ///
+    /// Typically, this field is set to 0 because COFF debugging information is deprecated.
+    /// [Source](https://0xrick.github.io/win-internals/pe4/#file-header-image_file_header).
+    // Q (JohnScience): Why is the name `number_of_symbol_table` and not `number_of_symbols`?
+    #[doc(alias("NumberOfSymbols"))]
     pub number_of_symbol_table: u32,
+    /// The size of the optional header, in bytes. This value should be zero for object files.
+    ///
+    /// The [`goblin::pe::optional_header::OptionalHeader`](crate::pe::optional_header::OptionalHeader) is meant to
+    /// represent either the 32-bit or the 64-bit optional header. The size of the optional header is used to determine
+    /// which one it is.
+    #[doc(alias("SizeOfOptionalHeader"))]
```

Comment:
what does this alias?

Reply:
@m4b When searching in docs for `SizeOfOptionalHeader`, it'll be found.

Here's the quote from the [documentation of `doc(alias = ".."`](https://doc.rust-lang.org/rustdoc/advanced-features.html#add-aliases-for-an-item-in-documentation-search)

> This feature allows you to add alias(es) to an item when using the rustdoc search through the doc(alias) attribute.

Reply:
ah interesting, i thought it aliased to the doc's of the type/expr in quotes (which doesn't exist); testing it locally, it does indeed work that way, neat, thank you, learned something new today!